app/services/discussion_service.rb

```diff
@@ -28,6 +28,12 @@ def self.create(discussion:, actor:)
     Events::NewDiscussion.publish!(discussion)
   end
 
+  def self.destroy(discussion:, actor:)
+    actor.ability.authorize!(:destroy, discussion)
+    discussion.comments.destroy_all
```

Comment:
Why destroy all the comments? I think one of the arguments for soft-deleting discussions is to allow for an 'undo' function, so we can bring it back later if there's a whoopsie.